apps/podcast-transcribe-episode/src/python/podcast_transcribe_episode/workflow_interface.py

```diff
@@ -0,0 +1,302 @@
+    backoff_coefficient=2,
+
+    # MaximumInterval specifies the maximum interval between retries. Useful for coefficients more than 1.
+    maximum_interval=timedelta(hours=2),
+
+    # MaximumAttempts specifies how many times to attempt to execute an Activity in the presence of failures. If this
+    # limit is exceeded, the error is returned back to the Workflow that invoked the Activity.
+
+    # We start off with a huge default retry count for each individual activity (1000 attempts * 2 hour max. interval
+    # = about a month worth of retrying) to give us time to detect problems, fix them, deploy fixes and let the workflow
+    # system just handle the rest without us having to restart workflows manually.
+    #
+    # Activities for which retrying too much doesn't make sense (e.g. due to the cost) set their own "maximum_attempts".
+    maximum_attempts=1000,
+
+    # NonRetryableErrorReasons allows you to specify errors that shouldn't be retried. For example retrying invalid
+    # arguments error doesn't make sense in some scenarios.
+    non_retryable_error_types=[
+
+        # Counterintuitively, we *do* want to retry not only on transient errors but also on programming and
+        # configuration ones too because on programming / configuration bugs we can just fix up some code or
+        # configuration, deploy the fixes and let the workflow system automagically continue on with the workflow
+        # without us having to dig out what exactly has failed and restart things.
+        #
+        # However, on "permanent" errors (the ones when some action decides that it just can't proceed with this
+        # particular input, e.g. process a story that does not exist) there's no point in retrying anything.
+        # anything anymore.
+        McPermanentError.__name__,
+
+    ],
+
+)
+"""
+Retry parameters.
+
+https://docs.temporal.io/docs/concept-activities/
+"""
+
+
+class AbstractPodcastTranscribeActivities(object):
+    """Activities interface."""
+
+    @classmethod
+    def _create_config(cls) -> PodcastTranscribeEpisodeConfig:
+        """
+        Create and return configuration instance to be used for running the workflow.
+
+        Might get overridden in case some configuration changes have to be made while running the tests.
+        """
+        return PodcastTranscribeEpisodeConfig()
+
+    def __init__(self):
+        super().__init__()
+        self.config = self._create_config()
+
+    @activity_method(
+        task_queue=TASK_QUEUE,
+
+        # ScheduleToStart is the maximum time from a Workflow requesting Activity execution to a worker starting its
+        # execution. The usual reason for this timeout to fire is all workers being down or not being able to keep up
+        # with the request rate. We recommend setting this timeout to the maximum time a Workflow is willing to wait for
+        # an Activity execution in the presence of all possible worker outages.
+        # schedule_to_start_timeout=None,
```

Comment:
Did you mean to comment this out?

Reply:
I did - here and elsewhere, it's just sample arguments that could be passed to `@workflow` (or however is it called) decorator plus copy-pasted descriptions of what do those arguments do.

I intended this workflow to serve as a reference for future workflows that we intend to write but maybe it's confusing to have all of this stuff here? Do they belong in `doc/workflows.markdown` instead? What do you think?

Reply:
I'd say include in `workflows.markdown`, as that will likely be the starting point for anyone trying to understand and implement a Temporal-related change for the first time; that said, there's no loss by keeping it in this file as well.

Reply:
Moved docs to `doc/workflows.markdown` as per your suggestion, added some initial samples for implementing / calling workflows too.